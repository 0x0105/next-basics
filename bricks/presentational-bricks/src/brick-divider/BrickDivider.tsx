import React from "react";
import { useTranslation } from "react-i18next";
import { NS_PRESENTATIONAL_BRICKS, K } from "../i18n/constants";
import { Divider } from "antd";

interface BrickDividerProps {
  type?: "horizontal" | "vertical";
  dashed?: boolean;
  dividerStyle?: any;
  dividerTitle?: string;
  orientation?: "center" | "left" | "right";
}

export function BrickDivider(props: BrickDividerProps): React.ReactElement {
  const { t } = useTranslation(NS_PRESENTATIONAL_BRICKS);

  return (
    <Divider
      {...{ type: props.type, dashed: props.dashed, style: props.dividerStyle, orientation: props.orientation }}
    >{props.dividerTitle}</Divider>
  );
}
