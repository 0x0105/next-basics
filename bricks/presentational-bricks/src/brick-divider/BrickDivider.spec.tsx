import React from "react";
import { shallow } from "enzyme";
import { BrickDivider } from "./BrickDivider";

describe("BrickDivider", () => {
  it("should work", () => {
    const wrapper = shallow(
      <BrickDivider type="horizontal" dashed={false} dividerStyle={{}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
