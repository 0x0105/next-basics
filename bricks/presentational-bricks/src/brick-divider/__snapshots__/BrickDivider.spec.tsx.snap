// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickDivider should work 1`] = `
<Divider
  dashed={false}
  style={Object {}}
  type="horizontal"
/>
`;
